### PR TITLE
fix DateOffset eq to depend on normalize attr

### PR DESCRIPTION
**Pull request by jbrockmendel**
This has a bullet point in #18854 but doesn't appear to have its own issue.

Current:
```
now = pd.Timestamp.now()
off = pd.offsets.Week(n=1, normalize=False)
normed = pd.offsets.Week(n=1, normalize=True)

>>> now + off == now + normed
False
>>> off == normed  # PR changes this comparison to False
True
```

- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: pandas/tests/tseries/offsets/test_offsets.py [hunk 1]
```diff
@@ -518,11 +518,10 @@ def setup_method(self, method):
...
     def test_different_normalize_equals(self):
-        # equivalent in this special case
-        offset = BDay()
-        offset2 = BDay()
-        offset2.normalize = True
-        assert offset == offset2
+        # GH#???? changed __eq__ to return False when `normalize` doesnt match
+        offset = self._offset()
+        offset2 = self._offset(normalize=True)
+        assert offset != offset2
```
Comment (jbrockmendel) [inline pandas/tests/tseries/offsets/test_offsets.py]: After substituting in `self._offset` for `BDay`, this test is identical to 5 others in this file.  I'm putting together a separate PR to try to de-duplicate and parametrize a bunch of things like this.

**Comment by gfyoung** [comment]
> This has a bullet point in #18854 but doesn't appear to have its own issue.

@jbrockmendel : How about this one here: https://github.com/pandas-dev/pandas/issues/17689?

**Review by gfyoung — APPROVED** [review]
This looks good to me!

cc @jreback

File: doc/source/whatsnew/v0.24.0.txt [hunk 2]
```diff
@@ -91,7 +91,7 @@ Categorical
 Datetimelike
 ^^^^^^^^^^^^
 
--
+- Fixed Bug where two :class:`DateOffset` objects with different `normalize` attributes could evaluate as equal (:issue:`21404`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.24.0.txt]: double backticks on normalize

**Review by jreback — CHANGES_REQUESTED** [review]
looks good. minor comment. ping on green.

**Comment by jreback** [comment]
I merged something which now causes a conflict here. @jbrockmendel can you rebase. @gfyoung can merge on green.

**Review by jreback — APPROVED** [review]

**Comment by gfyoung** [comment]
Thanks @jbrockmendel !

**Pull request merged**